fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PullUpPredicates.java

```diff
@@ -60,30 +67,84 @@
 public class PullUpPredicates extends PlanVisitor<ImmutableSet<Expression>, Void> {
 
     Map<Plan, ImmutableSet<Expression>> cache = new IdentityHashMap<>();
+    private final boolean getAllPredicates;
+
+    public PullUpPredicates(boolean all) {
+        getAllPredicates = all;
+    }
 
     @Override
     public ImmutableSet<Expression> visit(Plan plan, Void context) {
-        if (plan.arity() == 1) {
```

Comment:
why not remain this and avoid override too many plan visitor whose pullup is disabled?